x/transfer/ibc_handlers.go

```diff
@@ -96,41 +30,43 @@ func (im IBCModule) HandleAcknowledgement(ctx sdk.Context, packet channeltypes.P
 	if err != nil {
 		return errors.Wrapf(sdkerrors.ErrInvalidAddress, "failed to decode address from bech32: %v", err)
 	}
+	if !im.ContractManagerKeeper.HasContractInfo(ctx, senderAddress) {
```

Comment:
Transfer module can be used by both, either a regular account or a contract. In case it's a regular account we do not need to call sudo, we do early exit.
interchaintxs module can be used by contract account only, we dont need need to do early exit.

Reply:
don't we need to distribute fees before early exit?

Reply:
Regular user does not pay ibc fees, only contract does. Basically fees is a compensation for calling sudo handler by relayer. There is no callback, when origin of the ibc msg is regular account.